lib/Backend/GlobOpt.cpp

```diff
@@ -6935,20 +6960,34 @@ GlobOpt::SetSymStore(ValueInfo *valueInfo, Sym *sym)
     }
     if (valueInfo->GetSymStore() == nullptr || valueInfo->GetSymStore()->IsPropertySym())
     {
-        valueInfo->SetSymStore(sym);
+        SetSymStoreDirect(valueInfo, sym);
     }
 
     return sym;
 }
 
 void
+GlobOpt::SetSymStoreDirect(ValueInfo * valueInfo, Sym * sym)
+{
+    Sym * prevSymStore = valueInfo->GetSymStore();
+    if (prevSymStore && prevSymStore->IsStackSym() &&
+        prevSymStore->AsStackSym()->HasByteCodeRegSlot() &&
+        this->blockData.changedSyms)
```

Comment:
The thought is SetSymStore doesn't depend on whether changedSyms is available or not, so it just checks here. I will consider change it to assert or not in later change.

---
In reply to: [68113107](https://github.com/Microsoft/ChakraCore/pull/1152#discussion_r68113107) [](ancestors = 68113107)